src/cider/nrepl/middleware/inspect.clj

```diff
@@ -107,16 +110,30 @@
 (defn def-current-value [msg]
   (inspector-response msg (swap-inspector! msg inspect/def-current-value (symbol (:ns msg)) (:var-name msg))))
 
+(defn print-current-value-reply [{:keys [::print/print-fn session] :as msg}]
+  (let [inspector (-> session meta ::inspector)]
+    (with-open [writer (print/replying-PrintWriter :value msg msg)]
+      (binding [*print-length* (or *print-length* 100)
+                *print-level* (or *print-level* 20)]
+        ((or print-fn pprint/pprint) (:value inspector) writer)
+        (.flush writer)))
+    (respond-to msg :status :done)))
+
 (defn tap-current-value [msg]
   (inspector-response msg (swap-inspector! msg inspect/tap-current-value)))
 
 (defn tap-indexed [msg]
   (inspector-response msg (swap-inspector! msg inspect/tap-indexed (:idx msg))))
 
 (defn handle-inspect [handler {:keys [op inspect] :as msg}]
-  (if (and (= op "eval") inspect)
+  (cond
+    (and (= op "eval") inspect)
     (handle-eval-inspect handler msg)
 
+    (and (= op "inspect-print-current-value"))
```

Comment:
@alexander-yakushev I moved this out of `with-safe-transport` because I saw an exception in the `*nrepl-messages-buffer*`, which I think was caused by the new op sending mutiple lines vs the ops in `with-safe-transport` returing a response value. 

I didn't notice this exception before, because from the user's perspective both seemed to work.

With this change I don't see the exeception anymore. Can you double check that this ok?

Reply:
Yeah, I think it makes sense. This op is indeed different in this regard. It's fine to leave this as is, just leave a comment why it stands separately.